src/Mailgun/Model/Message/ShowResponse.php

```diff
@@ -142,7 +142,8 @@ public static function create(array $data)
             $response->setMessageUrl($data['message-url']);
         }
         if (isset($data['message-headers'])) {
-            $response->setMessageHeaders($data['message-headers']);
+            $headers = is_string($data['message-headers']) ? json_decode($data['message-headers']) : $data['message-headers'];
```

Comment:
$data['message-headers'] is always a string in the case that the message was received as a POST from the routes configured at [https://app.mailgun.com/app/routes](https://app.mailgun.com/app/routes).